### ISSUE TITLE
Bug when running cron

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_excimer#161]
No idea what this is about just hitting doing unrelated stuff

$ php admin/cli/adhoc_task.php -e
Server Time: Tue, 08 Feb 2022 15:41:28 +1100

Execute adhoc task: tool_testtasks\task\timed_adhoc_task
++ on_interval called with no profile ++
* line 73 of /admin/tool/excimer/classes/cron_manager.php: call to debugging()
* line 41 of /admin/tool/excimer/classes/cron_manager.php: call to tool_excimer\cron_manager::on_interval()
* line 689 of /lib/classes/date.php: call to tool_excimer\cron_manager::tool_excimer\{closure}()
* line 129 of /lib/classes/date.php: call to core_date::init_zones()
* line 272 of /lib/classes/date.php: call to core_date::normalise_timezone()
* line 2350 of /lib/moodlelib.php: call to core_date::get_user_timezone()
* line 327 of /calendar/type/gregorian/classes/structure.php: call to date_format_string()
* line 2292 of /lib/moodlelib.php: call to calendartype_gregorian\structure->timestamp_to_date_string()
* line 390 of /lib/cronlib.php: call to userdate()
* line 303 of /lib/cronlib.php: call to cron_set_process_title()
* line 198 of /lib/cronlib.php: call to cron_run_inner_adhoc_task()
* line 131 of /admin/cli/adhoc_task.php: call to cron_run_adhoc_tasks()
... started 15:41:28. Current memory use 18.6 MB.

**Comment by jaypha** [comment]
In order to get the cron profiling working, I needed to get the timer to trigger for every sample. It would be nice if the profiler
called a callback for every sample, but it doesn't. So I had to get the Timer to keep track of it. Sometimes it gets called early, and no data sampling has yet taken place. This only happens on the first timer event, and does not appear to affect the rest of the profiling.

I tried to put in a dummy timer to delay it about half a sample period to try to keep the event away from the profile event (ie. trigger in the middle of a sample period), but I couldn't get it to work.

So I put a debugging message to keep a pin on it, but otherwise left it alone.
